sway/commands/input/xkb_switch_layout.c

```diff
@@ -67,6 +73,9 @@ struct cmd_results *input_cmd_xkb_switch_layout(int argc, char **argv) {
 		relative = 0;
 	}
 
+	struct wl_list actions;
+	wl_list_init(&actions);
```

Comment:
We should be able to just use a `xkb_layout_index_t` array instead of a `wl_list` here

Reply:
Do you suggest to create an array of pre calculated xkb_layout_index_t[] and then zip it with original input devices list server.input->devices? This approach requires to do input device filtration twice. That's why I've introduced xkb_switch_layout_action struct. 

As I understand sway/wl doesn't provide vector-like type array only array of pointers so I still have to manually free() every single element.

Perhaps I didn't get your idea.

Reply:
Eh, right, I forgot that some filtering was necessary.

What I had in mind was something among these lines:

```c
struct xkb_switch_layout_action *actions = calloc(wl_list_length(&server.input->devices), sizeof(struct xkb_switch_layout_action));
size_t actions_len = 0;

// To add an entry:
actions[actions_len++] = (struct xkb_switch_layout_action){
    .keyboard = kbd,
    .layout = layout,
};

// To iterate:
for (size_t i = 0; i < actions_len; i++) {
    switch_layout(actions[i].keyboard, actions[i].layout);
}

// To cleanup:
free(actions);
```

Reply:
Fixed thanks!